src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -180,18 +191,21 @@ internal string AadAuthorityV2
 
             try
             {
-                BaseConfigurationManager effectiveConfigurationManager = GetEffectiveConfigurationManager(securityToken);
-                if (validationParameters.RefreshBeforeValidation)
-                    effectiveConfigurationManager.RequestRefresh();
-
-                BaseConfiguration configuration = await effectiveConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false);
-                string aadIssuer = configuration.Issuer;
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager(securityToken);
+                var baseConfiguration = await GetBaseConfiguration(effectiveConfigurationManager, validationParameters).ConfigureAwait(false);
+                string aadIssuer = baseConfiguration.Issuer;
 
                 if (!validationParameters.ValidateWithLKG)
                 {
                     if (IsValidIssuer(aadIssuer, tenantId, issuer))
                     {
-                        effectiveConfigurationManager.LastKnownGoodConfiguration = new OpenIdConnectConfiguration() { Issuer = aadIssuer };
+                        // We need to update LKG on the self-managed config manager.
+                        // The LKG of the one coming from provider, should be handled by the provider after successful validation
+                        // unless it is only fetched for the issuer and it's different than the authority.
+                        if (_configurationManagerProvider == null)
+                        {
+                            effectiveConfigurationManager.LastKnownGoodConfiguration = new OpenIdConnectConfiguration() { Issuer = aadIssuer };
```

Comment:
This is unfinished and needs more considerations. E.g. how should this be set in case of config fetched from provider? Shouldnt provider self manage this?